### DISCUSSION TITLE
How do i correctly raise an error using FastAPI's depeden

### DISCUSSION THREAD

**Discussion by ntoskrn** [fastapi/fastapi#9307] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends

app = FastAPI()

def validate_username(username: str):
    if "bad_word" in username:
        raise ValueError("Username contains a bad word")
    
    return username

@app.post("/users/{name}/register")
async def register_user(name: str = Depends(validate_username)):
    return {"username": name}
```

### Description

* Call the endpoint /users/bad_word_username/register
* Output should be 
```
{
  "detail": [
    {
      "loc": [
        "path",
        "name"
      ],
      "msg": "Username contains a bad word",
      "type": "value_error"
    }
  ]
}
```

* Real output: Internal server error

### Operating System

macOS

### FastAPI Version

0.95.0

### Python Version

Python 3.9.6

**Comment by mr-st0rm** [comment, marked as answer]
```
from fastapi import FastAPI, Depends
from fastapi.exceptions import HTTPException

app = FastAPI()

def validate_username(name: str):
    if "bad_word" in name:
        raise HTTPException(status_code=400, detail="Username contains a bad word")
    
    return name

@app.get("/users/{name}/register")
async def register_user(name: str = Depends(validate_username)):
    return {"username": name}
```

validating function args must be like a query args

**Reply by ntoskrn** [reply]
Thing is, I would rather want to get RequestValidationError than HTTPException, because RequestValidationError shows location, error type and message

**Reply by mr-st0rm** [reply]
no problem, u can use this error class from exceptions
